src/aws-cpp-sdk-core/include/aws/core/auth/signer/AWSAuthV4Signer.h

```diff
@@ -165,7 +165,7 @@ namespace Aws
             */
             bool PresignRequest(Aws::Http::HttpRequest& request, const char* region, const char* serviceName, long long expirationInSeconds = 0) const override;
 
-            virtual Aws::Auth::AWSCredentials GetCredentials(const Aws::Http::ServiceSpecificParameters &serviceSpecificParameters) const;
+            virtual Aws::Auth::AWSCredentials GetCredentials(const std::shared_ptr<Aws::Http::ServiceSpecificParameters> serviceSpecificParameters) const;
```

Comment:
I think that passing a shared_ptr by value is less efficient here than passing a reference to a shared_ptr

by value cons: memory allocation, shared_ptr atomic/locking refcount increase
by ref cons: double pointer deference in case it used later